lib/spack/spack/solver/asp.py

```diff
@@ -1524,7 +1532,9 @@ def key_fn(item):
                     DeclaredVersion(version=ver, idx=idx, origin=version_provenance.packages_yaml)
                 )
 
-        for spec in specs:
+        # enumerate so we can determine which list it came from
+        # cast to list to ensure addable types
```

Comment:
I don't understand the "cast to list" comment (also -- Python has no casting? I can google python and casting and get some results, but in Python you're really just constructing a list out of a set or whatever.  I'd avoid the word here)